paimon-core/src/main/java/org/apache/paimon/catalog/FileSystemCatalog.java

```diff
@@ -117,6 +119,9 @@ protected void dropTableImpl(Identifier identifier) {
 
     @Override
     public void createTableImpl(Identifier identifier, Schema schema) {
+        checkArgument(
+                !schema.options().containsKey(CoreOptions.PATH.key()),
```

Comment:
Can we just relax this check? For example, If it is the same path, we should allow it.

I am quite concerned that SHOW CREATE TABLE may not run properly.

Reply:
Thanks for reviewing. I would like to share my thoughts on the prohibition of setting table location in `FileSystemCatalog`. 

I think the `SHOW CREATE TABLE` command is usually used to create a test table with the same schema as the source table. Users typically use `SHOW CREATE TABLE` to get the DDL of a table and then create a test table with a different name for testing purposes. In this scenario, the location information in the DDL will inevitably mismatch the path assigned by the `FileSystemCatalog` for the new table, requiring users to modify the DDL in order to successfully create the table.

So even if the restrictions were relaxed, I believe it would still be somewhat confusing for users, as success would only be guaranteed when the specified location matches the one assigned by the catalog. If that's the case, why bother passing the location at all?

Therefore, instead of relaxing the checks here, I suggest we optimize the documentation by clearly stating this restriction in the documentation. We can declare the location management restrictions of `FileSystemCatalog` and the recommended usage of Spark DDL. What do you think?

Reply:
This is also our current approach to managing Iceberg tables in our platform. We host the location for all Iceberg tables on behalf of the users and strongly advise against specifying a location when creating tables. So far, this has been working well.

Reply:
I still think a relax check will be better. "create a test table with a different name for testing purposes", we can have very clear exception for this case.

Reply:
> Therefore, instead of relaxing the checks here, I suggest we optimize the documentation by clearly stating this restriction in the documentation. We can declare the location management restrictions of `FileSystemCatalog` and the recommended usage of Spark DDL. What do you think?

I'm +1 for this